src/tx_builder/mod.rs

```diff
@@ -410,31 +712,34 @@ pub fn balance_tx_capacity(
         {
             witnesses.push(Default::default());
         }
+        let mut ret_change_index = None;
         let new_tx = {
             let mut all_witnesses = tx.witnesses().into_iter().collect::<Vec<_>>();
             for (idx, witness_args) in &changed_witnesses {
                 all_witnesses[*idx] = witness_args.as_bytes().pack();
             }
             all_witnesses.extend(witnesses.clone());
+            let output_len = tx.outputs().len();
             let mut builder = tx
                 .data()
                 .as_advanced_builder()
                 .cell_deps(cell_deps.clone())
                 .inputs(inputs.clone())
                 .set_witnesses(all_witnesses);
             if let Some(output) = change_output.clone() {
+                ret_change_index = Some(output_len);
```

Comment:
Why can we assume the old change index is always the last output?

Reply:
1. next line it will add the change output to the builder, and the change cell's existence checked;
2. ret_change_index is a local variable of the loop, every loop it will be initialized to None;